packages/core-sdk/src/utils/feeUtils.ts

```diff
@@ -276,5 +277,20 @@ export const contractCallWithFees = async ({
     wallet,
     calls,
   });
-  return handleTxOptions({ rpcClient, txOptions, txHash });
+  return handleTransactionResponse(txHash, rpcClient, txOptions) as ContractCallWithFeesResponse<T>;
```

Comment:
all these forced casting need to be removed. Lets just change this to only support `Hash[]`. I feel like that is the simpliest approach.